visualizer/src/bin/etch_a_sketch.cpp

```diff
@@ -96,14 +96,16 @@ struct EtchaSketchPlayer : public audio::Player<EtchaSketchPlayer> {
         /* Get a handle to ourselves */
         EtchaSketchPlayer* self = (EtchaSketchPlayer*)userData;
 
+        self->serial_reciever.timestamp_contribution_logger.log_if_needed([self]() { return "timestamps current_trace_time " + std::to_string(self->current_trace_time) + " timestamp buffer_trace_time " + std::to_string(self->current_buffered_trace_time); });
+
         /* Guard on the case when we have zero pot value readings as there is nothing to play */
         if (self->serial_reciever.pot_num_values == 0) {
             std::fill_n(output, frameCount * 2, 0);
             return 0;
         }
 
         /* Guard on the case where the frequency is zero (ie. no note is being pressed on the keyboard) */
-        if (self->serial_reciever.trace_freq < 0.01) {
+        if (self->serial_reciever.trace_freq < 100) {
             std::fill_n(output, frameCount * 2, 0);
```

Comment:
todo: occasionally the trace contributions run off to infinity, this might some bug with what the arduino is sending us. This just low-pass filters the data for now